tests/view/RoomSettingsTest.java

```diff
@@ -0,0 +1,146 @@
+    ViewManagerModel viewManagerModel = new ViewManagerModel();
+
+    SwitchViewController switchViewController = SwitchViewUseCaseFactory.create(viewManagerModel);
+
+    LoginUserDataAccessInterface userDataAccessObject =
+        new LoginUserDataAccessInterface() {
+          @Override
+          public User getUser(String email, String password) {
+            return null;
+          }
+
+          @Override
+          public String getAccessToken(String email, String password) {
+            return null;
+          }
+
+          @Override
+          public DisplayUser getDisplayUser(String email) {
+            return null;
+          }
+        };
+    var roomsDataAccessObject =
+        new RoomsDataAccessInterface() {
+
+          public int deleteRoomCalls = 0;
+          public int changeRoomNameCalls = 0;
+
+          @Override
+          public Room getRoomFromId(
+              User user, LoginUserDataAccessInterface userDAO, String roomId) {
+            return null;
+          }
+
+          @Override
+          public Room addRoom(User user, LoginUserDataAccessInterface userDAO, String roomName) {
+            return null;
+          }
+
+          @Override
+          public void deleteRoom(User user, LoginUserDataAccessInterface userDAO, Room room) {
+            deleteRoomCalls++;
+          }
+
+          @Override
+          public void addUserToRoom(
+              User currentUser,
+              DisplayUser newUser,
+              LoginUserDataAccessInterface userDAO,
+              Room room) {}
+
+          @Override
+          public List<String> getAvailableRoomIds(User user) {
+            return null;
+          }
+
+          @Override
+          public void removeUserFromRoom(
+              User currentUser,
+              DisplayUser userToRemove,
+              LoginUserDataAccessInterface userDAO,
+              Room room) {}
+
+          @Override
+          public void changeRoomName(
+              User user, LoginUserDataAccessInterface userDAO, Room activeRoom, String roomName) {
+            changeRoomNameCalls++;
+            Assert.assertEquals("Test Room 2", roomName);
+          }
+        };
+
+    RoomSettingsViewModel roomSettingsViewModel = new RoomSettingsViewModel();
+    RoomsViewModel roomsViewModel = new RoomsViewModel();
+    RoomSettingsPresenter roomSettingsPresenter =
+        new RoomSettingsPresenter(roomsViewModel, viewManagerModel);
+    RoomSettingsInteractor roomSettingsInteractor =
+        new RoomSettingsInteractor(
+            roomsDataAccessObject, userDataAccessObject, roomSettingsPresenter);
+
+    RoomSettingsController roomSettingsController =
+        new RoomSettingsController(roomSettingsInteractor);
+    RoomSettingsView roomSettingsView =
+        new RoomSettingsView(roomSettingsViewModel, roomSettingsController, switchViewController);
+    views.add(roomSettingsView.contentPane, roomSettingsView.viewName);
+
+    JFrame jf = new JFrame();
```

Comment:
Is this necessary?

Reply:
This allows one to see the live results of tests, which is especially helpful when they fail (but also helpful for debugging)